flax/configurations.py

```diff
@@ -85,3 +85,8 @@ def static_bool_env(varname: str, default: bool) -> bool:
     name='use_orbax_checkpointing',
     default=False,
     help=('Whether to use Orbax to save checkpoints.'))
+
+flax_relaxed_naming = define_bool_state(
```

Comment:
Should we just set the default to True here?

Reply:
+1

Reply:
I'll give it a try - if it breaks too much I might have to do a staged off by default, we'll see soon.

Reply:
Flipping it on immediately is definitely going to break some code <-> checkpoint agreement in the wild, particularly for people using fancy config systems.  So we'll have to start with it off and alert some of our users first before defaulting to the new mode.